### PR TITLE
Clean up deadlock regression test

### PR DESCRIPTION
**Pull request by bghgary**
Clean up the deadlock regression test from #147.

## Changes

- Run entire test on a separate thread so gtest can detect deadlock via timeout
- Install hook after initialization (no hookEnabled flag needed)
- Hook only fires once (early return after first invocation) to prevent secondary deadlock from the hook sleeping on subsequent `wait()` calls
- Use plain `bool` for hookSignaled (only accessed by worker thread)
- Expanded comment on the sleep explaining non-determinism and its purpose
- Update comments and flow diagram to match code

## Dependency Updates

- Bump arcana.cpp GIT_TAG to b9bf9d8 (renamed `ARCANA_TESTING_HOOKS` to `ARCANA_TEST_HOOKS`, hooks moved to `arcana::test_hooks::blocking_concurrent_queue` namespace)
- Includes merge from main (#149 WorkQueue merge, #152 perf test flake fix)

## Verification

- **Fixed code (#151)**: 19/19 CI green — test passes
- **Broken code (#146)**: 12 deadlock failures as expected — test catches the bug

### CODE REVIEW

File: Tests/UnitTests/Shared/Shared.cpp [hunk 1]
```diff
@@ -131,82 +131,90 @@ TEST(Console, Log)
...
-            workerInHook.set_value();
-        }
-        if (hookEnabled.load())
-        {
+    // Run the full lifecycle on a separate thread so the gtest thread
+    // can detect a deadlock via timeout.
+    std::promise<void> testDone;
+    std::thread testThread([&]() {
+        auto runtime = std::make_unique<Babylon::AppRuntime>();
+
```
[the remark this answers is not in the document]
Comment (bghgary) [reply Tests/UnitTests/Shared/Shared.cpp]: [Responded by Copilot on behalf of @bghgary] The detach path only runs when the test has already failed (deadlock detected). The process is in a broken state at that point -- the leaked thread and stale references won't affect test results since FAIL() exits the test. Adding shared state for a failure-only path would add complexity without benefit.

File: Tests/UnitTests/Shared/Shared.cpp [hunk 2]
```diff
@@ -131,82 +131,90 @@ TEST(Console, Log)
...
+        // Wait for the worker to be in the hook (holding mutex, sleeping)
+        workerInHook.get_future().wait();
 
-    // Dispatch work and wait for completion
-    std::promise<void> ready;
-    runtime->Dispatch([&ready](Napi::Env) {
-        ready.set_value();
+        // Destroy — if the fix works, the destructor completes.
+        // If broken, it deadlocks and the timeout detects it.
+        runtime.reset();
```
[the remark this answers is not in the document]
Comment (bghgary) [reply Tests/UnitTests/Shared/Shared.cpp:206]: [Responded by Copilot on behalf of @bghgary] Intentional -- the outer 5s timeout on the gtest thread covers this case. If the hook never fires, workerInHook.wait() blocks, the gtest thread times out at 5s and detaches. The hook has always fired reliably in testing since it runs on every blocking_tick idle wait.

File: Tests/UnitTests/Shared/Shared.cpp [hunk 3]
```diff
@@ -131,82 +131,90 @@ TEST(Console, Log)
...
     if (status == std::future_status::timeout)
     {
-        destroyThread.detach();
-    }
-    else
-    {
-        destroyThread.join();
+        testThread.detach();
+        FAIL() << "Deadlock detected: AppRuntime destructor did not complete within 5 seconds";
     }
```
[the remark this answers is not in the document]
Comment (bghgary) [reply Tests/UnitTests/Shared/Shared.cpp:218]: [Responded by Copilot on behalf of @bghgary] Fair point on the message, but splitting into separate timeout phases adds complexity for a case that hasn't occurred in practice. The hook fires within milliseconds of being installed. If the 5s timeout fires, the destructor deadlock is by far the most likely cause.

File: Tests/UnitTests/Shared/Shared.cpp [hunk 4]
```diff
@@ -131,82 +131,90 @@ TEST(Console, Log)
...
+        // so the worker is idle and ready to enter the hook on the next wait.
+        std::promise<void> ready;
+        runtime->Dispatch([&ready](Napi::Env) {
+            ready.set_value();
+        });
+        ready.get_future().wait();
+
+        // Install the hook and dispatch a no-op to wake the worker,
+        // ensuring it cycles through the hook on its way back to idle.
+        arcana::set_before_wait_callback([&]() {
```
Comment (bghgary) [inline Tests/UnitTests/Shared/Shared.cpp]: Maybe consider adding a test_hooks::dispatcher namespace the arcana function to be more specific to dispatcher.

File: Tests/UnitTests/Shared/Shared.cpp [hunk 5]
```diff
@@ -131,82 +131,90 @@ TEST(Console, Log)
...
+            if (!hookSignaled)
+            {
+                hookSignaled = true;
+                workerInHook.set_value();
+            }
             std::this_thread::sleep_for(std::chrono::milliseconds(200));
```
Comment (bghgary) [inline Tests/UnitTests/Shared/Shared.cpp:197]: Add comment here to indicate that this technically is not deterministic and explain the purpose of this is to allow the runtime.reset to happen within this sleep, before the queued work starts.

**Review by bghgary — COMMENTED** [review]
Synchronous review comments.

**Review by bkaradzic-microsoft — APPROVED** [review]
YOLO!
